### PR TITLE
🦺 #203: map out consent wizard step 3

### PR DESCRIPTION
**Pull request by samrichca**
- add router for consent wizard step 3 get & post requests
- add zod schemas & jsdocs
- no unit tests required

### CODE REVIEW

File: packages/types/src/entities/ConsentResearchParticipation.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+export const ConsentResearchParticipationBase = z.object({
+	[ConsentQuestionId.enum.RESEARCH_PARTICIPATION__CONTACT_INFORMATION]: z.boolean(),
+	[ConsentQuestionId.enum.RESEARCH_PARTICIPATION__FUTURE_RESEARCH]: z.boolean(),
+});
+
+export const ConsentResearchParticipationRequest = ConsentResearchParticipationBase;
+export type ConsentResearchParticipationRequest = z.infer<
+	typeof ConsentResearchParticipationRequest
+>;
+export const ResearchParticipationRequestSchema: SchemaObject = generateSchema(
```
Comment (alyssadsouza) [inline packages/types/src/entities/ConsentResearchParticipation.ts:35]: definitely nit picking here but should this also be prefixed with `Consent` i.e. `ConsentResearchParticipationRequestSchema`? if not then for consistency I think I should change to match this in #221.

File: packages/types/tests/entities/ConsentResearchParticipation.ts
Comment (alyssadsouza) [inline packages/types/tests/entities/ConsentResearchParticipation.ts:1]: missing the `spec.ts` file extension - I ran the tests and was like ... shouldn't there be 21 tests...? 🕵️

File: packages/types/src/entities/ConsentResearchParticipation.ts (hunk 1 above)
Comment (samrichca) [reply packages/types/src/entities/ConsentResearchParticipation.ts]: not nitpicking at all, i thought i'd caught all of these. good eye!

File: packages/types/tests/entities/ConsentResearchParticipation.ts
Comment (samrichca) [reply packages/types/tests/entities/ConsentResearchParticipation.ts:1]: 23 now 🌽

**Review by alyssadsouza — COMMENTED** [review]
Looks good to me!

File: packages/types/tests/entities/ConsentResearchParticipation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	ConsentQuestionId,
+	ConsentResearchParticipationRequest,
+	ConsentResearchParticipationResponse,
+} from '../../src/entities/index.js';
+
+describe('ConsentResearchParticipationRequest', () => {
+	it('Must use boolean values', () => {
+		expect(
+			ConsentResearchParticipationRequest.safeParse({
+				[ConsentQuestionId.enum.RESEARCH_PARTICIPATION__CONTACT_INFORMATION]: true,
```
Comment (anncatton) [inline packages/types/tests/entities/ConsentResearchParticipation.spec.ts:32]: thanks for the separate tests for req/res

File: apps/consent-api/src/routers/consentResearchParticipation.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+ *         description: Forbidden. Provided Authorization token is valid but has insufficient permissions to make this request.
+ *       500:
+ *         description: Server error
+ */
+router.get('/', async (req, res) => {
+	logger.info(`GET /wizard/steps/consent-for-research-participation`);
+	// TODO: implement when auth layer is ready
+	try {
+		logger.info(`Retrieved Consent for Research Participation`);
+		res.status(200).send({
```
Comment (anncatton) [inline apps/consent-api/src/routers/consentResearchParticipation.ts:101]: same comment as above

File: packages/types/src/entities/ConsentResearchParticipation.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { z } from 'zod';
+import { generateSchema } from '@anatine/zod-openapi';
+import type { SchemaObject } from 'openapi3-ts/oas31';
+
+import { ConsentQuestionId } from './ConsentQuestion.js';
+
+export const ConsentResearchParticipationBase = z.object({
```
Comment (anncatton) [inline packages/types/src/entities/ConsentResearchParticipation.ts:26]: i think you mentioned this in the step 2 PR @alyssadsouza , about destructuring the enum - do you wanna use that in these prs for the `ConsentQuestionId` keys?

File: apps/consent-api/src/routers/consentResearchParticipation.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+ *         description: Forbidden. Provided Authorization token is valid but has insufficient permissions to make this request.
+ *       500:
+ *         description: Server error
+ */
+router.post('/', async (req, res) => {
+	logger.info(`POST /wizard/steps/consent-for-research-participation`);
+	// TODO: implement when auth layer is ready
+	try {
+		try {
+			const data = ConsentResearchParticipationRequest.parse(req.body);
```
Comment (anncatton) [inline apps/consent-api/src/routers/consentResearchParticipation.ts:64]: same feedback as in step 2, let's get rid of the nested try/catch and do an error type check to differentiate between a parsing error and a general "something went wrong" 500.

**Review by anncatton — CHANGES_REQUESTED** [review]
Mainly commenting on the try/catch in the `POST`, same as in step 2.

File: packages/types/src/entities/ConsentResearchParticipation.ts (hunk 4 above)
Comment (alyssadsouza) [reply packages/types/src/entities/ConsentResearchParticipation.ts:26]: yup good idea!

Comment (anncatton) [reply packages/types/src/entities/ConsentResearchParticipation.ts:29]: 🙏

**Review by anncatton — COMMENTED** [review]
this looks good to me, but we'll wait on step-2 being reviewed/merged

**Review by anncatton — APPROVED** [review]
lgtm 🚀 🎸 🪨

**Pull request merged**